client: send request via browser for web platform

toxeus opened the pull request:
Due to CORS policies cookie handling is forbidden in the browser. Therefore, let the browser handle the session cookies by using `package:http/browser_client.dart`.

Due to conditional imports the HTTP client from `package:http/http.dart` remains being used when run inside the dart VM.

This could be further cleaned up by isolating the cookie handling for the VM case only. Currently, this causes warnings to be printed in the browser console but otherwise works fine.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/inapay/bity_client?pullRequest=2) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/inapay/bity_client?pullRequest=2) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/inapay/bity_client?pullRequest=2) it.</sub>

toxeus committed:
client: send request via browser for web platform

Due to CORS policies cookie handling is forbidden
in the browser. Therefore, let the browser handle
the session cookies by using
`package:http/browser_client.dart`.

Due to conditional imports the HTTP client from
`package:http/http.dart` remains being used when
run inside the dart VM.

This could be further cleaned up by isolating
the cookie handling for the VM case only. Currently,
this causes warnings to be printed in the browser
console but otherwise works fine.

dspicher commented:
Would it be nicer if we just pass an http client as an argument? Could be helpful for testing, where we can pass a mock client.

toxeus commented:
I was considering that but came to the conclusion that it would unnecessary complicate this library. As it is now it behaves correctly for every supported platform and the user of this library does not need to be aware of the underlying issues with the different clients.
